packages/terra-tag/src/TagList.jsx

```diff
@@ -0,0 +1,329 @@
+  /**
+   * The string that labels the collection of tags, used in cases where the text label is not visible on
+   * the screen and required for minimum accessibility standards. Providing this prop adds the `aria-label`
+   * attribute to the Tag List container element. (Required)
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * If a visible text label is used with the collection of tags, provide a string of the ID for the html
+   * element containing the visible text label. Supplying the 'ariaLabelledBy' prop will override the 'ariaLabel'
+   * prop and adds the `aria-labelledby` attribute instead to the Tag List container element.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) For best practices, ensure the visible text
+   * in the html element `id` provided to the Tag List `ariaLabelledby` prop matches the text provided to the
+   * `ariaLabel` prop, for consistency in the case of fallback or errors.
+   */
+  ariaLabelledBy: PropTypes.string,
+  /**
+   * If additional visible information text is used, provide a string containing the IDs for html elements that
+   * help describe the intent of the group of tags.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * The content to be shown in the tags container.
+   */
+  children: PropTypes.node,
+  /**
+   * Indicates if the Tags container is rolled up or not.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  isCollapsible: false,
+};
+
+const TagList = (props) => {
+  const {
+    ariaLabel,
+    ariaLabelledBy,
+    ariaDescribedBy,
+    children,
+    intl,
+    isCollapsible,
+    ...customProps
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const [containerTabindex, setContainerTabindex] = useState('-1');
+  const [updatedCount, setUpdatedCount] = useState(React.Children.count(children));
+  const [rollUpCount, setRollUpCount] = useState(React.Children.count(children));
+  const [isCollapsed, setIsCollapsed] = useState(isCollapsible);
+  const currentTag = useRef(); // ID of the tag that will receive focus
+  const filterTagsRef = useRef();
+  const focusNode = useRef(0);
+  const isRollUpRemoved = useRef(false);
+  const containerHint = useRef(ariaLabel);
+
+  // Identifies the number of tags that needs to be hidden/rolled up
+  const generateRollUp = useCallback(() => {
+    const startIndex = TagsUtils.getRollUpIndex(filterTagsRef);
+    if (isCollapsed) {
+      setUpdatedCount(startIndex);
+      setRollUpCount(React.Children.count(children) - startIndex);
+    } else if (startIndex === React.Children.count(children)) {
+      setRollUpCount(0);
+    }
+  }, [children, isCollapsed]);
+
+  // Modifies the tabindex of the tag
+  const setTabIndex = (val) => {
+    const currentNode = currentTag.current ? currentTag.current : null;
```

Comment:
This line of code seems unnecessary.  You are using the value if defined or setting to null.  However, you just do an if check on the next line that nullifies the need for this line.

Reply:
Changes made Thanks.